File: 백준/Silver/10866. 덱/덱.cc
#include <bits/stdc++.h>
using namespace std;

int N, num;
string s;

int main(void) {
    ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);

    cin >> N;

    deque<int> dq;
    while(N--) {
        cin >> s;
        
        if(s == "push_front") {
            cin >> num;
            dq.push_front(num);
        }
        if(s == "push_back") {
            cin >> num;
            dq.push_back(num);
        }
        if(s == "pop_front") {
            if(dq.empty()) cout << -1 << '\n';
            else {cout << dq.front() << '\n'; dq.pop_front();}
        }
        if(s == "pop_back") {
            if(dq.empty()) cout << -1 << '\n';
            else {cout << dq.back() << '\n'; dq.pop_back();}
        }
        if(s == "front") {
            if(dq.empty()) cout << -1 << '\n';
            else {cout << dq.front() << '\n';}
        }
        if(s == "back") {
            if(dq.empty()) cout << -1 << '\n';
            else {cout << dq.back() << '\n';}
        }
        if(s == "size") {
            cout << dq.size() << '\n';
        }
        if(s == "empty") {
            cout << dq.empty() << '\n';
        }   
    }
}

File: 백준/Silver/10866. 덱/덱.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main(args: Array<String>) {

    val br = BufferedReader(InputStreamReader(System.`in`));
    val bw = BufferedWriter(OutputStreamWriter(System.out));

    val N = br.readLine().toInt();

    val dequeue = ArrayDeque<String>();

    (1..N).forEach() {
        val op = br.readLine().split(" ");

        if(op[0]=="push_front") {
            dequeue.addFirst(op[1]);
        }
        if(op[0]=="push_back") {
            dequeue.addLast(op[1]);
        }
        if(op[0]=="pop_front") {
            if(dequeue.isEmpty()) bw.write("-1\n");
            else {
                bw.write(dequeue.pollFirst()+"\n");
            }
        }
        if(op[0]=="pop_back") {
            if(dequeue.isEmpty()) bw.write("-1\n");
            else {
                bw.write(dequeue.pollLast()+"\n");
            }
        }
        if(op[0]=="size") {
            bw.write(dequeue.size.toString() + "\n");
        }
        if(op[0]=="empty") {
            if(dequeue.isEmpty()) bw.write("1\n");
            else bw.write("0\n");
        }
        if(op[0]=="front") {
            if(dequeue.isEmpty()) bw.write("-1\n");
            else bw.write(dequeue.peekFirst()+"\n");
        }
        if(op[0]=="back") {
            if(dequeue.isEmpty()) bw.write("-1\n");
            else bw.write(dequeue.peekLast()+"\n");
        }
    }

    br.close();
    bw.flush();
    bw.close();
}